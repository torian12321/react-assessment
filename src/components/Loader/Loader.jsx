import styled, { keyframes } from 'styled-components';

const rotationAnimation = keyframes`
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
`;

export const Loader = styled.div`
  color: #3970a3;
  font-size: 40px;
  width: 1em;
  height: 1em;
  margin: 16px auto;

  border: 6px solid currentColor;
  border-top-color: transparent;
  border-radius: 50%;
  animation: ${rotationAnimation} .8s ease infinite;
`;
