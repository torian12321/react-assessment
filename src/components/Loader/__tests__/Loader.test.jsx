import React from 'react';
import { render } from '@testing-library/react';
import { Loader } from '@components/Loader';
import 'jest-styled-components';

it('renders correctly', () => {
  const { container } = render(<Loader />);

  expect(container).toMatchSnapshot();
});
