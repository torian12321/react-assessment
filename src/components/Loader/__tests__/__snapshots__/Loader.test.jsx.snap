// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  color: #3970a3;
  font-size: 40px;
  width: 1em;
  height: 1em;
  margin: 16px auto;
  border: 6px solid currentColor;
  border-top-color: transparent;
  border-radius: 50%;
  -webkit-animation: eoUyJr .8s ease infinite;
  animation: eoUyJr .8s ease infinite;
}

<div>
  <div
    class="c0"
  />
</div>
`;
